src/go/rpk/pkg/cli/cmd/container/start.go

```diff
@@ -175,22 +326,26 @@ func startCluster(
 		return err
 	}
 
-	reqPorts := n * 5 // we need 5 ports per node
-	ports, err := vnet.GetFreePortPool(int(reqPorts))
-	if err != nil {
-		return err
-	}
-
 	// Start a seed node.
 	var (
-		seedID            uint
-		seedKafkaPort     = ports[0]
-		seedProxyPort     = ports[1]
-		seedSchemaRegPort = ports[2]
-		seedRPCPort       = ports[3]
-		seedMetricsPort   = ports[4]
+		seedID                                                         uint
+		seedKafkaPort                                                  = cPorts.KafkaPorts[0]
+		seedProxyPort, seedSchemaRegPort, seedRPCPort, seedMetricsPort uint
 	)
 
+	if len(cPorts.ProxyPorts) >= 1 {
+		seedProxyPort = cPorts.ProxyPorts[0]
+	}
+	if len(cPorts.RegistryPorts) >= 1 {
+		seedSchemaRegPort = cPorts.RegistryPorts[0]
+	}
+	if len(cPorts.BrokerRPCPorts) >= 1 {
+		seedRPCPort = cPorts.BrokerRPCPorts[0]
+	}
+	if len(cPorts.AdminPorts) >= 1 {
+		seedMetricsPort = cPorts.AdminPorts[0]
+	}
+
```

Comment:
Probably worth a validation that none of the ports in any slice overlap

Reply:
We can also require that all slices have the same length, which will simplify (eliminate) the conditionals that have been added below.

Reply:
I think I will just shift left and make sure that there is no overlap with the ports before passing them here